### PR TITLE
Added masking output support to tools->mask

### PR DESCRIPTION
**Pull request by kilroythethird**
Currently the mask tool outputs a combination of face, mask and masked face.
This PR add options to output only the mask or the masked face too.
Additionally it adds an option to output the whole frame.

This allows using the extraction and masking part of faceswap in an after effects workflow.
I am aware that this is not the goal of faceswap but since it became a pretty nice face extraction / clean and masking tool it would be nice to have.
This will become even more interesting as soon as better obstruction mask models are implemented.

### CODE REVIEW

File: tools/mask.py [hunk 1]
```diff
@@ -38,6 +38,8 @@ def __init__(self, arguments):
         self._input_is_faces = arguments.input_type == "faces"
         self._mask_type = arguments.masker
         self._output_opts = dict(blur_kernel=arguments.blur_kernel, threshold=arguments.threshold)
+        self._output_type = arguments.output_type
+        self._output_full_frame = arguments.full_frame
```
Comment (torzdf) [inline tools/mask.py]:
```py
        self._output_suffix = self._get_output_suffix()
```

Comment (torzdf) [reply tools/mask.py]: Then create function:
```py
    def _get_output_suffix(self):
        """ The filename suffix, based on selected output options

        Returns
        -------
        str:
            The suffix to be appended to the output filename
        """
        sfx = "{}_mask_preview_".format(self._mask_type)
        sfx += "face_" if not self._output_full_frame or self._input_is_faces else "frame_"
        sfx += "{}.png".format(self._output_type)
        return sfx
```

Comment (torzdf) [reply tools/mask.py]: then in function `_save`

change:
```py
        filename = os.path.join(self._saver.location,
                                "{}_{}_{}_mask_preview.png".format(os.path.splitext(frame)[0],
                                                                   idx,
                                                                   self._mask_type))
```
to:
```py
        filename = os.path.join(self._saver.location, "{}_{}_{}".format(os.path.splitext(frame)[0],
                                                                        idx,
                                                                        self._output_suffix))
```

**Comment by torzdf** [comment]
eg:
image

File: tools/mask.py (hunk 1 above)
Comment (torzdf) [reply tools/mask.py]: eg:
image

Comment (kilroythethird) [reply tools/mask.py]: Done. Thanks for the suggestion.
Definitely makes sense.

**Pull request merged**